### DISCUSSION TITLE
Avoid calling Cef.Shutdown or another workaround using both offscreen and winforms in the same app

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3581] [Q&A]
I create chromium webbrowsers without explicit call to Cef.initialize (using RequestContext to isolate browser instance cache). They are created in secondary threads of my app (from Thread pull). However when I close software I get the following message:
![image](https://user-images.githubusercontent.com/73235464/119327698-ee55ab80-bc8b-11eb-96bf-c9d012d55893.png)

I thought about just calling Cef initialize in main UI form constructor, however Cef initialize implies defining particular browser version (like Offscreen or Winforms), while I can create both types of browser during one app session. 

Any workarounds here?

**Comment by amaitland** [comment]
> Avoid calling Cef.Shutdown  

`Cef.Shutdown` must be called to ensure the `cache` is written to disk.

> I thought about just calling Cef initialize in main UI form constructo

That sounds fine.

> however Cef initialize implies defining particular browser version (like Offscreen or Winforms), while I can create both types of browser during one app session.

Use the `CefSharp.WinForms.CefSettings` instance if you are planning on using `WinForms` and `OffScreen`. 

`CefSharp.OffScreen.CefSettings` has audio muted by default. Basically just some different defaults.
